docs: Example - Integrating Crowdsec (#3651)
docker-mailserver/docker-mailserver: docs/content/examples/tutorials/crowdsec.md

---
title: 'Tutorials | Crowdsec'
---

> !!! quote "What is Crowdsec ?"

Reviewer: ```suggestion
!!! quote "What is Crowdsec?"
```

---
title: 'Tutorials | Crowdsec'
---

> !!! quote "What is Crowdsec ?"
>     
>     Crowdsec is an open source software that detects and blocks attackers using log analysis.
>     It has access to a global community-wide IP reputation database.
> 
>     [Source][crowdsec]
> 
>     [crowdsec]: https://www.crowdsec.net

Reviewer: ```suggestion
!!! quote "What is Crowdsec?"
    
    Crowdsec is an open source software that detects and blocks attackers using log analysis.
    It has access to a global community-wide IP reputation database.

    [Source](https://www.crowdsec.net)
```

## Installation

Crowdsec supports multiple [installation methods][crowdsec-installation-docs], however this page will use the docker installation.

### Docker mailserver

> In the docker-compose definition of the docker-mailserver, create a bind volume containing the logs.
> 
> ```yaml
> services:
>   mailserver:
>     image: ghcr.io/docker-mailserver/docker-mailserver:latest
>     container_name: mailserver
>     # Provide the FQDN of your mail server here (Your DNS MX record should point to this value)
>     hostname: mail.example.com
>     ports:
>       - "25:25"
>       - "465:465"
>       - "587:587"
>       - "993:993"
>     volumes:
>       - /docker-data/dms/mail-data/:/var/mail/
>       - /docker-data/dms/mail-state/:/var/mail-state/
>       - /docker-data/dms/mail-logs/:/var/log/mail/
>       - /docker-data/dms/config/:/tmp/docker-mailserver/
>       - /etc/localtime:/etc/localtime:ro
>     environment:
>       - ENABLE_RSPAMD=1
>       - ENABLE_CLAMAV=1
>       - ENABLE_FAIL2BAN=0
>     restart: always
> ```

Reviewer: I think you can minimize noise in the yaml example.

It will vary per user, so it's simpler to just focus on the addition you want to add while communicating the config path from the top-level key `services`.

````suggestion
In your `compose.yaml` for the DMS service, add a bind mount volume for `/var/log/mail`. This is to share the DMS logs to a separate crowdsec container.

!!! example

    ```yaml
    services:
      mailserver:
        volumes:
          - /docker-data/dms/mail-logs/:/var/log/mail/
    ```
````

### Crowdsec

> The crowdsec container needs to have access to the docker-mailserver logs.

Reviewer: I'm fine with the yaml `compose.yaml` snippet here as this may be less familiar to our docs readers.

```suggestion
The crowdsec container should also bind mount the same host path for DMS logs that was added in the DMS example above.
```

```yaml
services:
  image: crowdsecurity/crowdsec
  restart: unless-stopped
  ports:
    - "8080:8080"
    - "6060:6060"
  volumes:
    - /docker-data/dms/mail-logs/:/var/log/dms:ro
    - ./acquis.d:/etc/crowdsec/acquis.d
    - crowdsec-db:/var/lib/crowdsec/data/
  environment:

>     COLLECTIONS: crowdsecurity/postfix crowdsecutory/dovecot # These collection contains parsers and scenarios for postfix and dovecot

Reviewer: Typo for dovecot path? Might be better to shift the comment above?

```suggestion
    # These collection contains parsers and scenarios for postfix and dovecot
    COLLECTIONS: crowdsecurity/postfix crowdsecurity/dovecot
```

    TZ: Europe/Paris
volumes:
  crowdsec-db:
```

## Configuration

> Configure crowdsec to read and parse dms logs file.
> 
> Create the file `dms.yml` in `./acquis.d/`
> ```yaml
> ---
> source: file
> filenames:
>   - /var/log/dms/mail.log
> labels:
>   type: syslog
> ```

Reviewer: Uppercase DMS, optionally wrap example in the example admonition.

````suggestion
Configure crowdsec to read and parse the DMS log files.

!!! example

    Create the file `dms.yml` in `./acquis.d/`

    ```yaml
    ---
    source: file
    filenames:
      - /var/log/dms/mail.log
    labels:
      type: syslog
    ```
````
